Microsoft.Toolkit.Uwp.UI.Controls.Graph/PlannerTaskList/PlannerTaskList.cs

```diff
@@ -0,0 +1,346 @@
+            {
+                _add.Click -= Add_Click;
+            }
+
+            base.OnApplyTemplate();
+            if (GetTemplateChild(ControlTasks) is ListView list)
+            {
+                _list = list;
+                _list.ItemClick += List_ItemClick;
+                _list.Tapped += List_Tapped;
+                if (!IsWindowsPhone)
+                {
+                    _list.PreviewKeyUp += List_PreviewKeyUp;
+                }
+            }
+
+            if (GetTemplateChild(ControlAdd) is Button add)
+            {
+                _add = add;
+                _add.Click += Add_Click;
+            }
+
+            _input = GetTemplateChild(ControlInput) as TextBox;
+            if (MicrosoftGraphService.Instance.IsAuthenticated)
+            {
+                await LoadPlansAsync();
+            }
+            else
+            {
+                MicrosoftGraphService.Instance.IsAuthenticatedChanged -= Instance_IsAuthenticatedChanged;
+                MicrosoftGraphService.Instance.IsAuthenticatedChanged += Instance_IsAuthenticatedChanged;
+            }
+
+            if (IsWindowsPhone)
+            {
+                VisualStateManager.GoToState(this, MobileVisualState, false);
+            }
+        }
+
+        private async Task LoadPlansAsync()
+        {
+            try
+            {
+                MicrosoftGraphService graphService = MicrosoftGraphService.Instance;
+                await graphService.TryLoginAsync();
+                GraphServiceClient graphClient = graphService.GraphProvider;
+                IPlannerUserPlansCollectionPage plans = await graphClient.Me.Planner.Plans.Request().GetAsync();
+                Plans.Clear();
+                while (true)
+                {
+                    foreach (PlannerPlan plan in plans)
+                    {
+                        Plans.Add(plan);
+                    }
+
+                    if (plans.NextPageRequest == null)
+                    {
+                        break;
+                    }
+
+                    plans = await plans.NextPageRequest.GetAsync();
+                }
+
+                if (!string.Equals(InternalPlanId, PlanId))
+                {
+                    InternalPlanId = PlanId;
+                }
+            }
+            catch (Exception exception)
+            {
+                MessageDialog messageDialog = new MessageDialog(exception.Message);
+                await messageDialog.ShowAsync();
+            }
+        }
+
+        private async Task InitPlanAsync()
+        {
+            try
+            {
+                ClearTasks();
+                MicrosoftGraphService graphService = MicrosoftGraphService.Instance;
+                await graphService.TryLoginAsync();
+                GraphServiceClient graphClient = graphService.GraphProvider;
+                IPlannerPlanBucketsCollectionPage buckets = await graphClient.Planner.Plans[PlanId].Buckets.Request().GetAsync();
```

Comment:
Are we guaranteed that `graphClient.Planner.Plans[PlanId]` will never be null?

Reply:
@skendrot Yes, the `Plans` property isn't a local dictionary, this expression is to build a Graph query for remote api call, we wouldn't know if a plan exists or not until the method `GetAsync()` is called.